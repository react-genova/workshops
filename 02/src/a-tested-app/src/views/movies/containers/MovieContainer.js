// @flow
import * as React from 'react';
import MovieCollection from '../components/MovieCollection'
import type {MovieType} from '../components/types'


type Props = {
    movies: Array<MovieType>,
    onRate: (id:string, newRate:number) => void
};

const MovieContainer = ({movies,onRate}:Props) => 
    <MovieCollection movies={movies} onRate={onRate}/>
;

export default MovieContainer;