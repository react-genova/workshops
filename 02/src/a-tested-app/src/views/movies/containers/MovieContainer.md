### MovieContainer List ###

```js
const postRating=({id,rating})=>{alert(id+": "+rating);console.log(id,rating)};
const moviesFetch = {
    fulfilled: true,
    value: {
        "2": {
            id: "2",
            title: "Ave, Cesare!",
            original: "Hail, Caesar!",
            rating: 3,
            year: 2016
        },
        "3": {
            id: "3",
            title: "Frantic",
            original: "Frantic",
            rating: 3,
            year: 1988
        },
        "4": {
            id: "4",
            title: "Il ritorno dello Jedi",
            original: "Return of the Jedi",
            rating: 4,
            year: 1983
        }
    }
};

<MovieContainer moviesFetch={moviesFetch} postRating={postRating}/>
```

### MovieContainer Error ###

```js
const postRating=({id,rating})=>{alert(id+": "+rating);console.log(id,rating)};
const moviesRefresh=()=>{alert("refresh")}
const moviesFetch = {
    rejected: true
};

<MovieContainer moviesFetch={moviesFetch} moviesRefresh={moviesRefresh}/>
```

### MovieContainer pending ###

```js
const moviesFetch = {
    pending: true
};

<MovieContainer moviesFetch={moviesFetch} />
```