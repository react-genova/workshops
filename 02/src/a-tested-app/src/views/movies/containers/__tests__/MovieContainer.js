//@flow
import React from 'react';
import {_MovieContainer as MovieContainer} from '../MovieContainer';
import {PromiseState} from 'react-refetch'
import { shallow } from 'enzyme';

describe('MovieContainer component', () => {

    it('MovieContainer loading snapshot', () => {
        const moviesRefresh = jest.fn();
        const postRating = jest.fn();
        const moviesFetch = PromiseState.create();
        const component = shallow(
            <MovieContainer moviesFetch={moviesFetch} moviesRefresh={moviesRefresh} postRating={postRating}/>
        );
        expect(component).toMatchSnapshot();
    });

    it('MovieContainer error snapshot', () => {
        const moviesRefresh = jest.fn();
        const postRating = jest.fn();
        const moviesFetch = PromiseState.reject(new Error('boom'));
        const component = shallow(
            <MovieContainer moviesFetch={moviesFetch} moviesRefresh={moviesRefresh} postRating={postRating}/>
        );
        expect(component).toMatchSnapshot();
    });

    it('MovieContainer movies snapshot', () => {
        const moviesRefresh = jest.fn();
        const postRating = jest.fn();
        const movies = {
            "2": {
                id: "2",
                title: "Ave, Cesare!",
                original: "Hail, Caesar!",
                rating: 3,
                year: 2016
            },
            "3": {
                id: "3",
                title: "Frantic",
                original: "Frantic",
                rating: 3,
                year: 1988
            },
            "4": {
                id: "4",
                title: "Il ritorno dello Jedi",
                original: "Return of the Jedi",
                rating: 4,
                year: 1983
            }
        }
        const moviesFetch = PromiseState.resolve(movies);
        const component = shallow(
            <MovieContainer moviesFetch={moviesFetch} moviesRefresh={moviesRefresh} postRating={postRating}/>
        );
        expect(component).toMatchSnapshot();
    });

});