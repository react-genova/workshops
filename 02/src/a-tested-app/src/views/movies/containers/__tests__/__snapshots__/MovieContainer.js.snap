// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer component MovieContainer error snapshot 1`] = `
<Message
  negative={true}
>
  <MessageHeader>
    Errore
  </MessageHeader>
  <p>
    Errore durante il caricamento dei film
  </p>
  <Button
    as="button"
    onClick={[Function]}
  >
    Ricarica
  </Button>
</Message>
`;

exports[`MovieContainer component MovieContainer loading snapshot 1`] = `
<Loader
  active={true}
/>
`;

exports[`MovieContainer component MovieContainer movies snapshot 1`] = `
<MovieCollection
  movies={
    Array [
      Object {
        "id": "2",
        "original": "Hail, Caesar!",
        "rating": 3,
        "title": "Ave, Cesare!",
        "year": 2016,
      },
      Object {
        "id": "3",
        "original": "Frantic",
        "rating": 3,
        "title": "Frantic",
        "year": 1988,
      },
      Object {
        "id": "4",
        "original": "Return of the Jedi",
        "rating": 4,
        "title": "Il ritorno dello Jedi",
        "year": 1983,
      },
    ]
  }
  updateRate={[Function]}
/>
`;
